src/Nuvemshop/Provider.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace SocialiteProviders\Nuvemshop;
+
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * @see https://tiendanube.github.io/api-documentation/authentication
+ */
+class Provider extends AbstractProvider
+{
+    public const IDENTIFIER = 'NUVEMSHOP';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'read_content',
+        'write_content',
+        'read_products',
+        'write_products',
+        'read_customers',
+        'write_customers',
+        'read_orders',
+        'write_orders',
+        'read_coupons',
+        'write_coupons',
+        'write_scripts',
+        'write_shipping'
+    ];
+
+    /**
+     * The separating character for the requested scopes.
+     *
+     * @var string
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * Returns the configured user id that we're authenticating with
+     *
+     * @return string
+     */
+    private function getClientId()
+    {
+        return $this->getConfig('client_id');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(
+            sprintf(
+                'https://www.tiendanube.com/apps/%s/authorize',
+                $this->getClientId()
+            ),
+            $state
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://www.tiendanube.com/apps/authorize/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $accessToken = explode('.', $token)[0];
```

Comment:
Just use `[$bearerToken, $userId, $_] = explode('.', $token, 3);`

Reply:
Shouldn't it be [$bearerToken, $userId] = explode('.', $token, 2)? (it worked)

I tested it in my local environment and it returned me an error: "Undefined Array Key 2". Since I'm expecting a single string with both access_token and user_id separated by a dot, the limit parameter should be 2.

Reply:
I had the feeling that `$token` had the shape of a JWT.
Of couse, if the token contains only one separator dot, `explode('.', $token, 2)` is fine.